independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -274,8 +274,17 @@ public InputStream getResourceAsStream(String nm) {
                         data = handleTransform(name, data, transformers);
                     }
                     definePackage(name);
-                    return defineClass(name, data, 0, data.length,
-                            protectionDomains.computeIfAbsent(classPathElement, (ce) -> ce.getProtectionDomain(this)));
+                    try {
+                        return defineClass(name, data, 0, data.length,
+                                protectionDomains.computeIfAbsent(classPathElement, (ce) -> ce.getProtectionDomain(this)));
+                    } catch (LinkageError e) {
+                        //TODO: is this a good idea?
+                        //if we are already the initiating loader we might want to load this from the parent
+                        if (!parentFirst) {
+                            return parent.loadClass(name);
+                        }
```

Comment:
Is there a test or specific use-case this is fixing in our build or TS?

Reply:
This fixes some issues in the camel test suite that I am still investigating.

Reply:
I have removed this. The issue in Camel is that they are packaging parts of XML parsers, e.g. a jar contains a copy of QName but not the parser that is using it. This causes problems as the parser uses the JDK version of QName but the app sees the copy in the isolated ClassLoader. 

Once this is in I am planning on working with the camel team to resolve it.